armi/reactor/blocks.py

```diff
@@ -2344,36 +2318,40 @@ def autoCreateSpatialGrids(self):
                 )
             )
 
-        ringNumber = hexagon.numRingsToHoldNumCells(self.getNumPins())
+        nPins = self.getNumPins()
+        nRings = hexagon.numRingsToHoldNumCells(nPins)
         # For the below to work, there must not be multiple wire or multiple clad types.
         # note that it's the pointed end of the cell hexes that are up (but the
         # macro shape of the pins forms a hex with a flat top fitting in the assembly)
-        grid = grids.HexGrid.fromPitch(
-            self.getPinPitch(cold=True), numRings=0, cornersUp=True
-        )
-        spatialLocators = grids.MultiIndexLocation(grid=self.spatialGrid)
-        numLocations = 0
-        for ring in range(ringNumber):
-            numLocations = numLocations + hexagon.numPositionsInRing(ring + 1)
-        if numLocations != self.getNumPins():
+        if self.spatialGrid is not None:
+            grid = self.spatialGrid
+        else:
+            grid = grids.HexGrid.fromPitch(
+                self.getPinPitch(cold=True), numRings=nRings, cornersUp=True
+            )
+        spatialLocators = grids.MultiIndexLocation(grid=grid)
+        numLocations = hexagon.totalPositionsUpToRing(nRings)
+        if numLocations != nPins:
             raise ValueError(
-                "Cannot create spatialGrid, number of locations in rings{} not equal to pin number{}".format(
-                    numLocations, self.getNumPins()
-                )
+                f"Cannot create spatialGrid, number of locations in rings {nRings} "
```

Comment:
I like this.

But if this happens on a different node than the primary node, all the primary node log will see is this process was killed, not why.  Can we do something like this?

```python
msg = (
    f"Cannot create spatialGrid, number of locations in rings {nRings} "
    f"not equal to {nPins=}"
)
runLog.error(msg)
ValueError(msg)
```

Reply:
Will update. I've seen that pattern a few places, but not enough to know it was our preferred way to do logging / errors

Reply:
Well, long term, I would prefer to make it so `runLog.error()` did this by default.  After all, "errors" should kill a run.

And Mark and Nick agree with me.

But... it'll take time and a bunch of things will have to be changed/fixed downstream. So... not yet.